csharp/src/Ice/Protocol.cs

```diff
@@ -158,6 +159,45 @@ public static Ice.OutputStream CreateFailureResponseFrame(System.Exception excep
             return ostr;
         }
 
+        internal static Ice.Current CreateCurrent(int requestId, Ice.InputStream requestFrame,
+            Ice.ObjectAdapter adapter, Ice.Connection? connection = null)
+        {
+            int start = requestFrame.Pos;
+            var identity = new Ice.Identity(requestFrame);
+
+            // For compatibility with the old FacetPath.
+            string[] facetPath = requestFrame.ReadStringArray();
+            string facet;
+            if (facetPath.Length > 0)
+            {
+                if (facetPath.Length > 1)
+                {
+                    throw new Ice.MarshalException();
+                }
+                facet = facetPath[0];
+            }
+            else
+            {
+                facet = "";
+            }
+
+            string operation = requestFrame.ReadString();
+            byte mode = requestFrame.ReadByte();
+            var context = new Dictionary<string, string>();
+            int sz = requestFrame.ReadSize();
+            while (sz-- > 0)
+            {
+                string first = requestFrame.ReadString();
+                string second = requestFrame.ReadString();
+                context[first] = second;
+            }
+
+            var current = new Ice.Current(adapter, identity, facet, operation, (Ice.OperationMode)mode, context,
```

Comment:
directly return current no need to create this reference

Reply:
Fixed